addon/components/bs-modal.js

```diff
@@ -425,36 +429,34 @@ export default class Modal extends Component {
     this.addBodyClass();
     this.resize();
 
-    let callback = () => {
-      if (this.isDestroyed) {
+    this.inDom = true;
+
+    await this.handleBackdrop();
+
+    if (this.isDestroyed) {
+      return;
+    }
+
+    this.checkScrollbar();
+    this.setScrollbar();
+
+    schedule('afterRender', async () => {
```

Comment:
While we are at it, could we refactor this also to an async helper, similar to `nextRunloop()`, that we can await and then put the other code into the method body itself, rather than the callback?

I like these runloop-related helpers really, makes code much better readable, than the callback-based original primitives. Maybe extract into utils, or potentially an own addon? 😆